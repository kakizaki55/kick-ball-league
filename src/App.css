.App {
  display: block;
}
.routes {
  display: block;
}
.header {
  display: flex;
  justify-content: center;
}
.title {
  display: flex;
  justify-content: center;
}
