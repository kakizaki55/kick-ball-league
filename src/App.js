import './App.css';
import './views/TeamList/TeamsList';
import TeamsList from './views/TeamList/TeamsList';
import PlayerList from './views/PlayerList/PlayerList';
import PlayerDetail from './views/PlayerDetail/PlayerDetail';
import { BrowserRouter, NavLink, Switch, Route } from 'react-router-dom/cjs/react-router-dom.min';

function App() {
  return (
    <div className="App">
      <h1 className="title">kick ball</h1>
      <BrowserRouter>
        <header className="header">
          <NavLink to="/">Home</NavLink>
          <NavLink to="/teams">Teams</NavLink>
          <NavLink to="/players">Players</NavLink>
        </header>
        <div className="routes">
          <Switch>
            <Route exact path="/">
              <h1>Kick Ball League Home Page</h1>
            </Route>
            <Route exact path="/teams/:id" component={PlayerList} />
            <Route exact path="/teams" component={TeamsList} />
            <Route exact path="/players/:id" component={PlayerDetail} />
            <Route exact path="/players" component={PlayerList} />
          </Switch>
        </div>
      </BrowserRouter>
    </div>
  );
}

export default App;
