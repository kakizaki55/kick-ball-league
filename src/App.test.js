import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

it('finds kickball', async () => {
  const { container } = render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );

  screen.getByText('kick ball');

  expect(container).toMatchSnapshot();
});
