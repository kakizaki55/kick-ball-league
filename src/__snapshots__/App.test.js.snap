// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finds kickball 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      class="title"
    >
      kick ball
    </h1>
    <header
      class="header"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/teams"
      >
        Teams
      </a>
      <a
        href="/players"
      >
        Players
      </a>
    </header>
    <div
      class="routes"
    >
      <h1>
        Kick Ball League Home Page
      </h1>
    </div>
  </div>
</div>
`;
